src/timer.rs

```diff
@@ -82,18 +82,17 @@ impl TimerExt<SYST> for SYST {
 impl Periodic for Timer<SYST> {}
 
 /// Trait for general purpose timer peripherals
-pub trait GeneralPurposeTimer {
+pub trait GeneralPurposeTimer: Enable + Reset {
     type MasterMode;
 
-    /// Enables the timer peripheral with the peripheral bus
-    fn enable(&mut self, rcc: &mut Rcc);
     /// Selects the master mode for this timer
     fn select_master_mode(&mut self, variant: Self::MasterMode);
 }
 
 impl<T: GeneralPurposeTimer> Timer<T> {
-    pub fn new(mut tim: T, rcc: &mut Rcc) -> Self {
-        tim.enable(rcc);
+    pub fn new(tim: T, rcc: &mut Rcc) -> Self {
+        T::enable(rcc);
+        T::reset(rcc);
```

Comment:
Note: At first glance, this looks like a reset was added, but the reset was already present in the previous `enable` method. Looks correct to me.